Framework/API/inc/MantidAPI/AlgorithmHasProperty.h

```diff
@@ -1,18 +1,13 @@
 #ifndef MANTID_API_ALGORITHMHASPROPERTY_H_
 #define MANTID_API_ALGORITHMHASPROPERTY_H_
 
-//------------------------------------------------------------------------------
-// Includes
-//------------------------------------------------------------------------------
 #include "MantidAPI/DllConfig.h"
 #include "MantidKernel/TypedValidator.h"
+#include <boost/make_shared.hpp>
```

Comment:
`boost::make_shared()` seems to be needed only by `AlgorithmHasProperty::clone()` here. Why not move its definition to the .cpp file?